cloud/parameterstore/parameter_record.go

```diff
@@ -0,0 +1,22 @@
+package parameterstore
+
+import (
+	"context"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// ParameterRecord stores metadata information about parameters. This never
+// stores the value of the parameter itself.
+type ParameterRecord struct {
+	// Name is the unique full path identifier for the parameter.
+	Name string `bson:"_id" json:"_id"`
```

Comment:
Using `Name` as the variable name with `bson:"_id" json:"_id"` is a bit confusing. Can we either change the variable name to ID or change the bson tags to name? I couldn't find a precedent for this anywhere else in the codebase so I think it's better to avoid it unless there's a strong reason why it needs to be set up like this.

Reply:
I guess we could rename it to ID, but then things could be a little confusing since the name _is_ the Parameter Store identifier, and I want it to be clear that the Parameter Store name _is_ the unique identifier. Having it be the document's `_id` is important so that it's the document's primary key and therefore automatically has a unique index to ensure that no two documents can have the same name. Also, if none of the BSON tags on the data model are `_id`, the doc will automatically be assigned an ObjectId as its `_id`, which I'd prefer not to have.